warehouse/manage/views/__init__.py

```diff
@@ -527,12 +523,8 @@ def totp_provision(self):
 
         totp_secret = self.user_service.get_totp_secret(self.request.user.id)
         if totp_secret:
-            self.request.session.flash(
-                "Account cannot be linked to more than one authentication "
-                "application at a time",
-                queue="error",
-            )
-            return HTTPSeeOther(self.request.route_path("manage.account"))
+            # User wants to replace their existing TOTP application.
+            self.request.session.clear_totp_secret()
```

Comment:
Is clearing the secret from the session here strictly necessary? I would expect us to be able to remove this entire section.

Reply:
Strictly necessary, no, but I think it's the right behavior. If we don't clear the session here, the user will still be able to view the QR code and update their TOTP settings, however they would end up with the same TOTP secret as they had before. I think it's best to change to a new TOTP secret because the reason the user might be replacing their TOTP is because they lost their old phone and don't want someone else to be able to use it to authenticate as them. I can update the comment in the code explaining this more.

Reply:
Thanks, it was unclear to me that this secret will be used to render the page if it's available, I agree it should be cleared. Made a slight simplification to just always clear this before rending the page instead.